pennylane/ops/op_math/parametrized_hamiltonian.py

```diff
@@ -0,0 +1,172 @@
+    term must be initialized with concrete parameters to obtain an Operator.
+    >>> coeffs = [2, f1, f2]
+    >>> obs = [qml.PauliX(0)@qml.PauliX(1), qml.PauliY(0)@qml.PauliY(1), qml.PauliZ(0)@qml.PauliZ(1)]
+    >>> H = ParametrizedHamiltonian(coeffs, obs)
+    >>> H.H_fixed
+      2*(PauliX(wires=[0]) @ PauliX(wires=[1]))
+    >>> H.H_parametrized([2.5, 3.6], t)
+      (0.5984721441039564*(PauliY(wires=[0]) @ PauliY(wires=[1]))) + (1.9450883011253033*(PauliZ(wires=[0]) @ PauliZ(wires=[1])))
+
+    The resulting object can be passed parameters, and will return an Operator representing the
+    ParametrizedHamiltonian with the specified parameters:
+
+    >>> H([1.2, 2.3], 0.5)
+    (2*(PauliX(wires=[0]) @ PauliX(wires=[1]))) + ((0.5646424733950354*(PauliY(wires=[0]) @ PauliY(wires=[1]))) + (2.0184398923478573*(PauliZ(wires=[0]) @ PauliZ(wires=[1]))))
+
+    """
+
+    def __init__(self, coeffs, observables):
+
+        if qml.math.shape(coeffs)[0] != len(observables):
+            raise ValueError(
+                "Could not create valid Hamiltonian; "
+                "number of coefficients and operators does not match."
+            )
+
+        # ToDo: assuming this is how we want to do things, make this clear in the documentation
+        # since organizing into H_fixed and H_parametrized potentially moves around the order of the terms compared
+        # to how the terms were entered, sorting seems the least ambiguous organization for the wires
+        self.wires = Wires.all_wires([op.wires for op in observables], sort=True)
+
+        self._ops = list(observables)
+        self._coeffs = coeffs
+
+        self.H_coeffs_fixed = []
+        self.H_coeffs_parametrized = []
+        self.H_ops_fixed = []
+        self.H_ops_parametrized = []
+
+        for coeff, obs in zip(coeffs, observables):
+            if callable(coeff):
+                self.H_coeffs_parametrized.append(coeff)
+                self.H_ops_parametrized.append(obs)
+            else:
+                self.H_coeffs_fixed.append(coeff)
+                self.H_ops_fixed.append(obs)
+
+    def __call__(self, params, t, wires=None):
+        H = self.H_fixed() + self.H_parametrized(params, t)
+
+        if wires:
+            wire_map = dict(zip(self.wires, wires))
+            H = qml.map_wires(H, wire_map)  # ToDo: is this the behaviour we want?
+
+        return H
+
+    def __repr__(self):
+        return f"ParametrizedHamiltonian: terms={qml.math.shape(self.coeffs)[0]}"
+
+    def H_fixed(self):
+        """The fixed term(s) of the ParametrizedHamiltonian. Returns a qml.Sum operator of qml.SProd operators
+        (or a single qml.SProd operator in the event that there is only one term in H_drift)."""
+        if self.H_coeffs_fixed:
+            return qml.ops.dot(self.H_coeffs_fixed, self.H_ops_fixed)  # pylint: disable=no-member
+        return 0
+
+    def H_parametrized(self, params, t):
```

Comment:
Since this will be in the documentation, could we add the usual parts of the docstrings like the arguments, what's returned, etc.?